registry/storage/blob_test.go

```diff
@@ -273,12 +344,12 @@ func TestLayerUploadZeroLength(t *testing.T) {
 		t.Fatalf("unexpected number of bytes copied: %v > 0", nn)
 	}
 
-	dgst, err := digest.FromReader(bytes.NewReader([]byte{}))
+	dgst, err := digest.FromReader(bytes.NewReader(blob))
 	if err != nil {
 		t.Fatalf("error getting zero digest: %v", err)
```

Comment:
May want to add a message argument for when this fails, since this is no longer only writing zero digest.